app/livechat/client/views/app/tabbar/customerChatHistory.js

```diff
@@ -0,0 +1,230 @@
+import {Mongo ,MongoInternals} from 'meteor/mongo';
+import { Tracker } from 'meteor/tracker';
+import { ReactiveDict } from 'meteor/reactive-dict';
+import { Session } from 'meteor/session';
+
+const ITEMS_COUNT = 50;
+let  historyResult 
+let len, msgs
+let allHistoryChat ;
+
+Template.customerChatHistory.helpers({
+	
+	isAllChat:function(){
+        // will return is have to load all chat 
+		return Template.instance().isAllChat.get();
+	},
+	isChatClicked:function(){
+		// will return that if you have clicked in a single chatHistory
+		return Template.instance().isChatClicked.get();
+	},
+	isfound:function() {
+		// will return if find any search result
+		var isfound= Session.get('found');
+	
+		return isfound;
+	},
+	isLoading() {
+		return Template.instance().isLoading.get();
+	},
+	searchResults(){
+		// will return search result
+		var r = Session.get('searchResult');
+		return 	Template.instance().searchResult.get();
+	},
+	
+	previousChats() {
+		// will return pervious chats list
+		let history = Template.instance().history.get();
+		let newHisTory = []
+		for(i=1; i<history.length; i++){
+			var d = Date.parse(history[i].ts);
+			var D = new Date();
+			D.setTime(d);
+			var newTs = D.toTimeString();
+			var time = getTime(newTs);
+			history[i].time = time;
+			history[i].count = history[i].msgs-3;
+			newHisTory.push(history[i]);
+		}
+		Template.instance().modifiedHistory.set(newHisTory);
+		return Template.instance().modifiedHistory.get();
+	},
+	len(){
+		// will return length of total messages in room
+		len = Template.instance().historyResu.get();
+		return len = len.length-2;
+	},
+	title() {
+		
+		let title = moment(this.ts).format('L LTS');
+
+		if (this.label) {
+			title += ` - ${ this.label }`;
+		}
+		return title;
+	},
+});
+
+Template.customerChatHistory.onCreated(function() {
+	const currentData = Template.currentData();
+	this.visitorId = new ReactiveVar();
+	this.isLoading = new ReactiveVar(false);
+	this.modifiedHistory = new ReactiveVar([])
+	this.history = new ReactiveVar([]);
+	this.offset = new ReactiveVar(0);
+	this.total = new ReactiveVar(0);
+	this.isAllChat = new ReactiveVar(true);
+	this.isChatClicked = new ReactiveVar(true);
+	this.autorun(async () => {
+		const { room } = await APIClient.v1.get(`rooms.info?roomId=${ currentData.rid }`);
+		if (room && room.v) {
+			this.visitorId.set(room.v._id);
+		}
+	});
+
+	this.autorun(async () => {
+		
+		if (!this.visitorId.get() || !currentData || !currentData.rid) {
+			return;
+		}
+
+		const offset = this.offset.get();
+	
+		this.isLoading.set(true);
+		const { history, total } = await APIClient.v1.get(`livechat/visitors.chatHistory/room/${ currentData.rid }/visitor/${ this.visitorId.get() }?count=${ ITEMS_COUNT }&offset=${ offset }`);
+		
+		this.isLoading.set(false);
+		this.total.set(total);
+		this.history.set(this.history.get().concat(history));
+		// this will load all the chat history chat messages
+		allHistoryChat = [];
```

Comment:
Why do you need this code? You don't need to fetch all of the chat histories once, you can rely on the pagination and you'll fetch more rooms on the `scroll` event until we identify that you fetched all of the chat histories.

Reply:
fixed